examplescripts/ubuntu_deploy.py

```diff
@@ -1,123 +1,202 @@
+"""
+This script is for deploying two ubuntu containers and testing sshing between them
+Steps:
+- creating new pool
+- loading network with name `management` to create containers on it or creating new network with same name if it doesn't exist
+- deploy network with access node
+- deploy two containers on the network
+- use ssh forwarding to ssh between the containers
+"""
+
 from jumpscale.loader import j
 from time import sleep
-import uuid
+import netaddr
 import random
 
-zos = j.sals.zos
+
+def wait_workload(workload_id, expiry=10):
+    expiry = expiry or 10
+    expiration_provisioning = j.data.time.now().timestamp + expiry * 60
+    # wait for workload
+    while True:
+        workload = zos.workloads.get(workload_id)
+        if workload.info.result.workload_id:
+            success = workload.info.result.state.value == 1
+            if not success:
+                error_message = workload.info.result.message
+                msg = f"Workload {workload.id} failed to deploy due to error {error_message}. For more details: {j.core.identity.me.explorer_url}/reservations/workloads/{workload.id}"
+                print(msg)
+            return success
+        if expiration_provisioning < j.data.time.get().timestamp:
+            j.sals.reservation_chatflow.solutions.cancel_solution([workload_id])
+        sleep(2)
+    return False
+
+
+def get_node_free_ip(ip_range):
+    """
+    get a node free ip, skips the first ip as it is assigned to the network
+    """
+    hosts = netaddr.IPNetwork(ip_range).iter_hosts()
+    next(hosts)
+    return str(next(hosts))  # skip ip used by node
```

Comment:
this doesn't really get a free IP. it will always return `10.100.x.2`. to get a free IP you need to check all workloads deployed on the node and attached to the target network and exclude their IPs from the node range.